tests/trainers/java/files_and_directories.java

```diff
@@ -0,0 +1,185 @@
+/**
+Test to check the Java File module.
+TODO
+*/
+import java.io.*;
+import java.io.IOException;
+import java.util.UUID;
+
+class Test {
+    int step = 1;
+    String textContent = "Content of the text file";
+    int errors = 0;
+
+    private String createTextFile(String filepath) {
+        try {
+            FileWriter file = new FileWriter(filepath);
+            file.write(this.textContent);
+            file.close();
+            return "OK";
+        } catch (IOException ex) {
+            return "KO - " + ex.getMessage();
+        }
+    }
+
+    private String readTextFile(String filepath) {
+        try {
+            FileReader file = new FileReader(filepath);
```

Comment:
I would move the FileReader outside, so that you can use a `finally` block

```suggestion
        FileReader file = null;
        try {
            FileReader file = new FileReader(filepath);
            // snip
        } catch (IOException ex) {
            // snip
        } finally {
            if (file != null) {
                file.close();
            }
        }
```

Otherwise the file handle would not be closed in case e.g. an `IOException` is raised while doing the read.

Reply:
In this case this is only a style question, so feel free to also ignore this comment if you feel it's not applicable.